modules/op-event-pruner-observability.adoc

```diff
@@ -0,0 +1,77 @@
+// This module is included in the following assemblies:
+// * install_config/customizing-configurations-in-the-tektonconfig-cr.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="event-pruner-observability_{context}"]
+= Observability metrics of the event-based pruner
+
+:FeatureName: The event-based pruner
+include::snippets/technology-preview.adoc[]
+
+The event-based pruner exposes detailed metrics through the `tekton-pruner-controller` controller `Service` definition on port `9090` in OpenTelemetry format for monitoring, troubleshooting, and capacity planning.
+
+Following are categories of the metrics exposed:
+
+* Resource processing
+* Performance timing
+* State tracking
+* Error monitoring
+
+Most pruner metrics use labels to provide additional context. You can use these labels in PromQL{nbsp}queries or dashboards to filter and group the metrics.
+
+[cols="1,3"options="header"]
+|===
+| Label | Description
+
+| `namespace`
+| The Kubernetes namespace of the `PipelineRun` or `TaskRun`.
+
+| `resource_type`
+| The Tekton resource type.
+
+| `status`
+| The outcome of processing a resource.
+
+| `operation`
+| The pruning method that deleted a resource.
+
+| `reason`
+| Specific cause for skipping or error outcomes.
+|===
+
+== Resource processing metrics
+
+[cols="2,3,1,3,2,4",options="header"]
+|===
+| Category | Name | Type | Description | Labels | Sample
+| Resource processing | `tekton_pruner_controller_resources_processed_total` | Counter | Total resources processed | namespace, resource_type, status | `tekton_pruner_controller_resources_processed_total{instance="tekton-pruner-controller.tekton-pipelines.svc.cluster.local:9090", job="tekton-pruner", namespace="default", otel_scope_name="tekton_pruner_controller", resource_type="pipelinerun", status="success"}`
+| Resource processing | `tekton_pruner_controller_resources_deleted_total` | Counter | Total resources deleted | namespace, resource_type, operation | `tekton_pruner_controller_resources_deleted_total{instance="tekton-pruner-controller.tekton-pipelines.svc.cluster.local:9090", job="tekton-pruner", namespace="default", operation="ttl", otel_scope_name="tekton_pruner_controller", resource_type="pipelinerun"}`
+|===
+
+
+== Performance timing metrics
+
+[cols="2,3,1,3,2,2",options="header"]
+|===
+| Category | Name | Type | Description | Labels | Bucket 
```

Comment:
Apply feedback for this section as per the feedback provided in the Resource processing metrics section.